httpx/_utils.py

```diff
@@ -67,6 +68,8 @@ def primitive_value_to_str(value: "PrimitiveData") -> str:
         return "false"
     elif value is None:
         return ""
+    elif isinstance(value, enum.Enum):
+        return str(value.value)
     return str(value)
```

Comment:
I think I'd be happier if HTTPX had an extra branch `if isinstance(value, str): return value` here, and then raised an explicit `TypeError` for all remaining cases. 

For `Enum`, we could _assume_ that people want `value.value` to be used, but why assume and potentially cause unexpected behavior when we can make users provide exactly what they want with a few keystrokes?

Is my thinking, at least

Reply:
The `if isinstance(value, str): return value` solves the issue that I want to solve here.

Should I proceed? Aren't we going to have some side effects?

Reply:
It was kind of what happened on https://github.com/encode/httpx/pull/2523/, and we had to revert.